terraform/modules/ingress/certificates.tf

```diff
@@ -0,0 +1,179 @@
+  subject {
+    country     = "France"
+    common_name = "client"
+    # organization = "127.0.0.1"
+  }
+}
+
+resource "tls_locally_signed_cert" "ingress_client_certificate" {
+  count                 = length(tls_cert_request.ingress_client_cert_request)
+  cert_request_pem      = tls_cert_request.ingress_client_cert_request.0.cert_request_pem
+  ca_private_key_pem    = tls_private_key.client_root_ingress.0.private_key_pem
+  ca_cert_pem           = tls_self_signed_cert.client_root_ingress.0.cert_pem
+  validity_period_hours = "168"
+  allowed_uses = [
+    "key_encipherment",
+    "digital_signature",
+    "server_auth",
+    "client_auth",
+    "any_extended",
+  ]
+}
+
+resource "pkcs12_from_pem" "ingress_client_pkcs12" {
+  count           = length(tls_locally_signed_cert.ingress_client_certificate)
+  password        = ""
+  cert_pem        = tls_locally_signed_cert.ingress_client_certificate.0.cert_pem
+  private_key_pem = tls_private_key.ingress_client_private_key.0.private_key_pem
+  ca_pem          = tls_self_signed_cert.client_root_ingress.0.cert_pem
+}
+
+resource "local_sensitive_file" "ingress_ca" {
+  count           = length(tls_self_signed_cert.root_ingress)
+  content         = tls_self_signed_cert.root_ingress.0.cert_pem
+  filename        = "${path.root}/.terraform/${var.container_name}/server/ca.crt"
+  file_permission = "0644"
+}
+
+resource "local_sensitive_file" "ingress_crt" {
+  count           = length(tls_locally_signed_cert.ingress_certificate)
+  content         = tls_locally_signed_cert.ingress_certificate.0.cert_pem
+  filename        = "${path.root}/.terraform/${var.container_name}/server/ingress.crt"
+  file_permission = "0644"
+}
+
+resource "local_sensitive_file" "ingress_key" {
+  count           = length(tls_private_key.ingress_private_key)
+  content         = tls_private_key.ingress_private_key.0.private_key_pem
+  filename        = "${path.root}/.terraform/${var.container_name}/server/ingress.key"
+  file_permission = "0644"
+}
+
+resource "local_sensitive_file" "ingress_client_ca" {
+  count           = length(tls_self_signed_cert.client_root_ingress)
+  content         = tls_self_signed_cert.client_root_ingress.0.cert_pem
+  filename        = "${path.root}/.terraform/${var.container_name}/client/client_ca.crt"
+  file_permission = "0644"
+}
+
+resource "local_sensitive_file" "ingress_client_crt" {
+  count           = length(tls_locally_signed_cert.ingress_client_certificate)
+  content         = tls_locally_signed_cert.ingress_client_certificate.0.cert_pem
+  filename        = "${path.root}/.terraform/${var.container_name}/client/client.crt"
+  file_permission = "0644"
+}
+
+resource "local_sensitive_file" "ingress_client_key" {
+  count           = length(tls_private_key.ingress_client_private_key)
+  content         = tls_private_key.ingress_client_private_key.0.private_key_pem
+  filename        = "${path.root}/.terraform/${var.container_name}/client/client.key"
+  file_permission = "0644"
```

Comment:
Keys should not be 644 but 600

Reply:
Well, I've had a lot of issues with that. Because we're using nginx unpriviledged, mounting the file directly in docker keeps its permissions, and so the server can't access it if it's not 644